darts/models/forecasting/prophet.py

```diff
@@ -44,54 +67,267 @@ def __init__(self,
             Some optional keyword arguments for Prophet.
             For information about the parameters see:
             `The Prophet source code <https://github.com/facebook/prophet/blob/master/python/prophet/forecaster.py>`_.
-
         """
 
         super().__init__()
 
+        self._auto_seasonalities = self._extract_auto_seasonality(prophet_kwargs)
+
+        self._add_seasonalities = dict()
+        add_seasonality_calls = add_seasonalities if isinstance(add_seasonalities, list) else [add_seasonalities]
+        for call in add_seasonality_calls:
+            self._store_add_seasonality_call(seasonality_call=call)
+
         self.country_holidays = country_holidays
-        self.freq = frequency
         self.prophet_kwargs = prophet_kwargs
         self.model = None
 
     def __str__(self):
         return 'Prophet'
 
-    def fit(self, series: TimeSeries):
-        super().fit(series)
+    def fit(self, series: TimeSeries, future_covariates: Optional[TimeSeries] = None):
+        super().fit(series, future_covariates)
         series = self.training_series
 
-        in_df = pd.DataFrame(data={
+        fit_df = pd.DataFrame(data={
             'ds': series.time_index,
             'y': series.univariate_values()
         })
 
-        # TODO: user-provided seasonalities, or "auto" based on stepduration
         self.model = prophet.Prophet(**self.prophet_kwargs)
-        if self.freq is not None:
-            if series.freq_str in ['MS', 'M', 'ME']:
-                interval_length = 30.4375
-            elif series.freq_str == 'Y':
-                interval_length = 365.25
-            else:
-                interval_length = pd.to_timedelta(series.freq_str).days
-            self.model.add_seasonality(name='custom', period=self.freq * interval_length,
-                                       fourier_order=5)
-
-        # Input built-in country holidays
+
+        # add user defined seasonalities (from model creation and/or pre-fit self.add_seasonalities())
+        if self._add_seasonalities:
```

Comment:
Is this `if` needed? It looks like `self._add_seasonalities` dict always exists.

Reply:
If the dict is empty, then this condition is not met. Was more to highlight that covariates are only used if they were added to the model.